### PR TITLE
[Doc] Update allocator service & multi-cluster allocation documentation

### PR DESCRIPTION
**Pull request by pooneh-m**
**What type of PR is this?**

/kind documentation

**What this PR does / Why we need it**:
Add a note in the doc for (1) priority number interpretation in multi-cluster; (2) emphasize on configuring cert-manager `certificate` based on the underlying system; (3) prereq for cmd-line tool installation in allocator doc.

**Which issue(s) this PR fixes**:

Closes https://github.com/googleforgames/agones/issues/1781 and https://github.com/googleforgames/agones/issues/1582

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/docs/Advanced/multi-cluster-allocation.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ The remainder of this article describes how to enable multi-cluster allocation.
 ## Define Cluster Priority
 
 {{< ghlink href="pkg/apis/multicluster/v1/gameserverallocationpolicy.go" >}}GameServerAllocationPolicy{{< /ghlink >}} is the CRD defined by Agones for setting multi-cluster allocation rules. In addition to cluster priority, it describes the connection information for the target cluster, including the game server namespace, agones-allocator endpoint and client K8s secrets name for redirecting the allocation request. Here is an example of setting the priority for a cluster and it's connection rules. One such resource should be defined per cluster. For clusters with the same priority, the cluster is chosen with a probability relative to its weight.
+The lower the `priority` number is the higher is the corresponding cluster's priority for allocating game servers.
```
Comment (roberthbailey) [inline site/content/en/docs/Advanced/multi-cluster-allocation.md:20]: Nit: This doesn't quite read correctly. How about:


In addition to cluster priority, it describes the connection information for the target cluster, including the game server namespace, agones-allocator endpoint and client K8s secrets name for redirecting the allocation request. Game servers will be allocated from clusters with the lowest `priority` number. If there are no available game servers available in clusters with the lowest `priority` number, they will be allocated from clusters with the next lowest `priority` number. For clusters with the same priority, the cluster is chosen with a probability relative to its weight. 

Here is an example of setting the priority for a cluster and it's connection rules. One such resource should be defined per cluster.

File: site/content/en/docs/Advanced/allocator-service.md [hunk 2]
```diff
@@ -35,7 +36,10 @@ If the `agones-allocator` service is installed as a `LoadBalancer` [using a stat
 
 Replace the default server TLS certificate with a certificate with CN and subjectAltName. There are multiple approaches to generate a certificate. Agones recommends using [cert-manager.io](https://cert-manager.io/) solution for cluster level certificate management.
 
-In order to use cert-manager solution, first, [install cert-manager](https://cert-manager.io/docs/installation/kubernetes/) on the cluster. Then, [configure](https://cert-manager.io/docs/configuration/) an `Issuer`/`ClusterIssuer` resource and last configure a `Certificate` resource to manage allocator-tls `Secret`.
+In order to use cert-manager solution, first, [install cert-manager](https://cert-manager.io/docs/installation/kubernetes/) on the cluster.
```
Comment (roberthbailey) [inline site/content/en/docs/Advanced/allocator-service.md:39]: As long as you are updating this paragraph, can you make it into proper sentences?

e.g. "In order to use the cert-manager solution, first [install cert-manager] on the cluster. Then [configure] a ..."

File: site/content/en/docs/Advanced/multi-cluster-allocation.md (hunk 1 above)
Comment (pooneh-m) [reply site/content/en/docs/Advanced/multi-cluster-allocation.md]: Good suggestion. Thanks!

File: site/content/en/docs/Advanced/allocator-service.md (hunk 2 above)
Comment (pooneh-m) [reply site/content/en/docs/Advanced/allocator-service.md]: Done. Thanks!

File: site/content/en/docs/Advanced/allocator-service.md [hunk 3]
```diff
@@ -35,15 +36,18 @@ If the `agones-allocator` service is installed as a `LoadBalancer` [using a stat
...
+Make sure to configure the `Certificate` based on your system's requirements, including the validity `duration`.
 
 Here is an example of using a self-signed `ClusterIssuer` for configuring allocator-tls `Secret`:
 
 ```bash
 #!/bin/bash
 # Create a self-signed ClusterIssuer
 cat <<EOF | kubectl apply -f -
-apiVersion: cert-manager.io/v1alpha2
+apiVersion: cert-manager.io/v1
```
Comment (roberthbailey) [inline site/content/en/docs/Advanced/allocator-service.md:50]: Can you ack that it is ok to publish these changes for the existing release (vs. gating them on the next agones release)? I think that it should be ok, but want to double check.

**Review by roberthbailey — APPROVED** [review]

**Comment by roberthbailey** [comment]
And I submitted without getting an ack.... oops.

File: site/content/en/docs/Advanced/allocator-service.md (hunk 3 above)
Comment (pooneh-m) [reply site/content/en/docs/Advanced/allocator-service.md:50]: They are ok, unless a cluster is using an old version of cert-manager, which is independent to Agones version.
I manually tested it on Agones v1.6 cluster.

**Pull request merged**